libraries/cbor/src/reader.rs

```diff
@@ -38,9 +38,8 @@ pub enum DecoderError {
 }
 
 /// Deserialize CBOR binary data to produce a single [`Value`], expecting that there is no additional data.
-/// Supports arbitrarily nested CBOR (so the [`DecoderError::TooMuchNesting`] error is never emitted).
 pub fn read(encoded_cbor: &[u8]) -> Result<Value, DecoderError> {
-    read_nested(encoded_cbor, None)
+    read_nested(encoded_cbor, Some(i8::MAX))
```

Comment:
Actually, now I'm wondering why we even need to support infinite nesting. Shouldn't we just have `read_nested(i8)`? When is infinite nesting useful? We could keep `read()` for convenience but I'm not even sure it's a good decision. It's better to see explicitly at call site the nesting. Users could wrap that value in a convenience `read()` function used everywhere in their project.)

Reply:
Yeah, IIRC the `read()` / `write()` entrypoints were for [convenience](https://github.com/google/OpenSK/pull/329#discussion_r653451639), so that library users mostly don't have to think about it (after all, 127 levels of nesting ought to be enough for anyone?).

Reply:
Intention was to keep the simple interface from before. Means we have to decide between the current state of this PR (default 127, `None` possible) and have users always pass in the maximum nesting?

I start to see the appeal of the simpler interface that @ia0 is proposing. It does require most users to write a wrapper function for their preferred nesting level, but that is already true for everyone who doesn't like the 127 default. Also, if we don't think anyone should ever call with `None`, removing that `Option` makes sense too. And we don't have to provide the default 127 anymore.

So this PR good to merge, but feel free to change to the simple API too if you'd like.